src/core/lwt_result.mli

```diff
@@ -0,0 +1,54 @@
+(* Lightweight thread library for OCaml
+ * http://www.ocsigen.org/lwt
+ * Interface Lwt
+ * Copyright (C) 2005-2008 Jérôme Vouillon
+ * Laboratoire PPS - CNRS Université Paris Diderot
+ *               2009-2012 Jérémie Dimino
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, with linking exceptions;
+ * either version 2.1 of the License, or (at your option) any later
+ * version. See COPYING file for details.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+ * 02111-1307, USA.
+ *)
+
+(** Module [Lwt_result]: explicit error handling *)
+
+(** This module provides helpers for values of type [('a, 'b) result Lwt.t] *)
+
+type (+'a, +'b) t = ('a, 'b) Result.result Lwt.t
+
+val return : 'a -> ('a, _) t
+
+val fail : 'b -> (_, 'b) t
+
+val lift : ('a, 'b) Result.result -> ('a, 'b) t
+
+val ok : 'a Lwt.t -> ('a, _) t
+
+val map : ('a -> 'b) -> ('a,'e) t -> ('b,'e) t
+
+val map_err : ('e1 -> 'e2) -> ('a,'e1) t -> ('b,'e2) t
+
+val bind : ('a,'e) t -> ('a -> ('b,'e) t) -> ('b,'e) t
```

Comment:
It would be nice to have this and `bind_lwt` in both `Ok` and `Error` forms for conditionally rewriting each case specifically.